src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -221,5 +235,97 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
         }
         
     }
+    
+	@Override
+	public void doRouteStatusChange(DocumentHeader header) {
+		super.doRouteStatusChange(header);
+		VendorDetail vendorDetail = (VendorDetail) getBusinessObject();
+		WorkflowDocument workflowDoc = header.getWorkflowDocument();
+		boolean existingPaymentWorksVendor = getPaymentWorksVendorService().isExistingPaymentWorksVendorByDocumentNumber(this.getDocumentNumber());
+
+		if (workflowDoc.isFinal()) {
```

Comment:
Sorry, but I haven't seen an answer to this question, so I will ask again, "do we want this to happen on final or processed docs?"